lib/statsd.rb

```diff
@@ -14,14 +14,77 @@
 #   statsd = Statsd.new('localhost').tap{|sd| sd.namespace = 'account'}
 #   statsd.increment 'activate'
 class Statsd
-  class RubyUdpClient
-    attr_reader :key, :sock
+  class BufferedUDPClient < UDPClient
+    @max_packet_size = 511
+    (class << self; attr_accessor :max_packet_size; end)
 
-    def initialize(address, port, key = nil)
+    def initialize(address, port)
+      super(address, port)
+      @buffer = String.new
+    end
+
+    def flush!
+      return unless @buffer.bytesize > 0
+      sock.write(@buffer)
+      @buffer.clear
```

Comment:
Hmpf. No luck here. Our only options are writing a C extension or eating the needles GC churn. Thoughts?